src/main/java/randoop/generation/ComponentManager.java

```diff
@@ -60,6 +65,15 @@ public class ComponentManager {
    */
   private final Collection<Sequence> gralSeeds;
 
+  /** For each scope in the SUT, statistics about its constants. */
+  public ScopeToConstantStatistics scopeToConstantStatistics = new ScopeToConstantStatistics();
+
+  /**
+   * Cache for constant sequences filtered by scope and type. Key format: scopeKey + ":" +
```

Comment:
Rather than including the scope key in this cache, does it make sense to move the cache within ConstantStatistics instead?
And if that makes sense, does it make sense to have ConstantStatistics segregate sequences by type
(That is, its current `Map<Sequence, ConstantUses>` data structure could be changed to `Map<Type, Map<Sequence, ConstantUses>>`.  Are most/all uses going to filter by type anyway?  I suspect so, and if so it makes more sense to filter once, early than to filter on every access.  That might also eliminate the need for a cache.)